src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import clsx from 'clsx'
+import { range } from 'lodash-es'
+import Button from 'components/button'
+import Select from 'components/select'
+import { ChevronLeftOutline, ChevronRightOutline } from '../icons'
+import PageButton from './components/page-button'
+
+/**
+ * Pagination component properties
+ */
+export type PaginationProps = {
+  /** Default initial page number */
+  defaultCurrent?: number
+  /** Total number of data items */
+  total: number
+  /** Determine whether to show pageSize select */
+  showSizeChanger?: boolean
+  /** Determine whether you can jump to pages directly */
+  showQuickJumper?: boolean
+}
+
+/**
+ * A long list can be divided into several pages using Pagination, and only one page will be loaded at a time.
+ *
+ * ### When To Use
+ *
+ * * When it will take a long time to load/render all items.
+ * * If you want to browse the data by navigating through pages.
+ */
+
+export function Pagination({
+  defaultCurrent = 1,
+  total,
+  showSizeChanger = false,
+  showQuickJumper = false,
+}: PaginationProps) {
+  const [totalPages, setTotalPages] = useState(Math.ceil(total / 10))
```

Comment:
10 is again a hard-code magic number. We should either set at a `prop` or a constant value.